### PR TITLE
Feature/comment specific line diffs

### PR DESCRIPTION
**Pull request by baristaGeek**
## Description
This PR uses the console log detection function to comment on the specific diff in the specific file containing the console log. 

To do so, we tell OpenAI to also return the code line itself, and then get the code line in the position of the diff (not the actual file) which is what Octokit needs. 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)  
- [x] New feature (non-breaking change which adds functionality)  
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)  
- [ ] Documentation update  
- [ ] Chore: cleanup/renaming, etc  
- [ ] RFC  

## Acceptance
- [x] I have read [the Contributing guidelines](CONTRIBUTING.md)
- [x] I have read the [Code of Conduct](CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test_hello_worlds/hello.rs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+fn main() {
```
Comment (baristaGeek) [inline test_hello_worlds/hello.rs:1]: remove this console log - normal octokit

Comment (baristaGeek) [inline test_hello_worlds/hello.rs:1]: remove this console log - normal octokit

Comment (baristaGeek) [inline test_hello_worlds/hello.rs:1]: remove this console log - programmatic filename

File: test_hello_worlds/python.cpp [hunk 2]
```diff
@@ -0,0 +1 @@
+print("This line will be printed.")
```
Comment (baristaGeek) [inline test_hello_worlds/python.cpp:1]: remove this console log - programmatic filename

File: test_hello_worlds/hello.rs (hunk 1 above)
Comment (baristaGeek) [inline test_hello_worlds/hello.rs:1]: remove this console log - recent sha

File: test_hello_worlds/python.cpp (hunk 2 above)
Comment (baristaGeek) [inline test_hello_worlds/python.cpp:1]: remove this console log - recent sha

Comment (baristaGeek) [inline test_hello_worlds/python.cpp:1]: remove this console log - old sha

Comment (baristaGeek) [inline test_hello_worlds/python.cpp:1]: remove this console log - old sha

Comment (baristaGeek) [inline test_hello_worlds/python.cpp:1]: remove this console log - old sha

File: test_hello_worlds/hello.rs (hunk 1 above)
Comment (baristaGeek) [inline test_hello_worlds/hello.rs:1]: Remove console log - todays sha hardcoded filenmae

Comment (baristaGeek) [inline test_hello_worlds/hello.rs:1]: Remove console log - todays sha hardcoded filenmae

Comment (baristaGeek) [inline test_hello_worlds/hello.rs:1]: Remove console log - todays sha hardcoded filenmae

Comment (baristaGeek) [inline test_hello_worlds/hello.rs:1]: Remove console log - todays sha hardcoded filenmae

File: test_hello_worlds/python.cpp (hunk 2 above)
Comment (baristaGeek) [inline test_hello_worlds/python.cpp:1]: Remove console log - c1a toString

File: test_hello_worlds/hello.rs (hunk 1 above)
Comment (baristaGeek) [inline test_hello_worlds/hello.rs:1]: Remove console log - c1a toString

File: test_hello_worlds/python.cpp (hunk 2 above)
Comment (baristaGeek) [inline test_hello_worlds/python.cpp:1]: response.data.head.sha

File: test_hello_worlds/hello.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+console.log("hello world");
```
Comment (baristaGeek) [inline test_hello_worlds/hello.ts:1]: response.data.head.sha

File: test_hello_worlds/hello.cpp [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#include <iostream>
```
Comment (baristaGeek) [inline test_hello_worlds/hello.cpp:1]: response.data.head.sha

File: test_hello_worlds/hello.rs (hunk 1 above)
Comment (baristaGeek) [inline test_hello_worlds/hello.rs:1]: response.data.head.sha

File: test_hello_worlds/hello.ts (hunk 3 above)
Comment (baristaGeek) [inline test_hello_worlds/hello.ts:1]: Please remove this console log

File: test_hello_worlds/hello.rs (hunk 1 above)
Comment (baristaGeek) [inline test_hello_worlds/hello.rs:1]: Please remove this console log

File: test_hello_worlds/python.cpp (hunk 2 above)
Comment (baristaGeek) [inline test_hello_worlds/python.cpp:1]: Please remove this console log

File: test_hello_worlds/hello.cpp (hunk 4 above)
Comment (baristaGeek) [inline test_hello_worlds/hello.cpp:1]: This file contains at least one console log. Please remove any present. - additions.length

File: test_hello_worlds/hello.rs (hunk 1 above)
Comment (baristaGeek) [inline test_hello_worlds/hello.rs:1]: This file contains at least one console log. Please remove any present. - additions.length

File: test_hello_worlds/python.cpp (hunk 2 above)
Comment (baristaGeek) [inline test_hello_worlds/python.cpp:1]: This file contains at least one console log. Please remove any present. - additions.length

File: test_hello_worlds/hello.ts (hunk 3 above)
Comment (baristaGeek) [inline test_hello_worlds/hello.ts:1]: This file contains at least one console log. Please remove any present. - additions.length

File: utils/actions/detectConsoleLogs.ts [hunk 5]
```diff
@@ -122,19 +142,47 @@ export default async function detectConsoleLogs({
...
+                  const latestCommitHash = result.data.head.sha;
+
+                  return octokit.request(
+                    "POST /repos/{owner}/{repo}/pulls/{pull_number}/reviews",
+                    {
+                      owner,
+                      repo,
+                      pull_number: issue_number,
+                      commit_id: latestCommitHash,
+                      event: "REQUEST_CHANGES",
```
Comment (EstebanDalelR) [inline utils/actions/detectConsoleLogs.ts:168]: Ooof, do we want to request changes? 
If this is the only available option, yes, but we want to give a heads up.
Sometimes I commit console logs to view the prod results!

File: utils/actions/detectConsoleLogs.ts [hunk 6]
```diff
@@ -67,6 +68,22 @@ function getAdditions(filePatch: string) {
...
 
+function getConsoleLogPosition(filePatchAndIndividualLine: any) {
+  let positionInDiff = 1;
+  const { filePatch, individualLine } = filePatchAndIndividualLine;
+
+  // get the position of the indiviudalLine in th filePatch
+  const lines = filePatch.split("\n");
+  for (let i = 0; i < lines.length; i++) {
+    if (lines[i].includes(individualLine)) {
+      positionInDiff = i + 1;
```
Comment (EstebanDalelR) [inline utils/actions/detectConsoleLogs.ts:79]: So the position in diff is 1 indexed?

File: utils/actions/detectConsoleLogs.ts [hunk 7]
```diff
@@ -107,7 +124,10 @@ export default async function detectConsoleLogs({
     Python comments start with #.
     Other console functions such as console.info() shouldn't be counted as console logs.
     Ignore code comments from this analysis. 
-    If there is a console log, print "true", else print "false"`;
+    If there is a console log, return "true", else return "false".
+    If you return true, return a string that that has 2 values: result (true) and the line of code.
+    The line value, is the actual line in the file that contains the console log.
+    For example: true,console.log("hello world");`;
```
Comment (EstebanDalelR) [inline utils/actions/detectConsoleLogs.ts:130]: Do we _really_ need to use GPT for this?
Wouldn't a list of console logs per lang be faster and less prone to failing?

File: utils/actions/detectConsoleLogs.ts (hunk 6 above)
Comment (baristaGeek) [reply utils/actions/detectConsoleLogs.ts:79]: yes

File: utils/actions/detectConsoleLogs.ts (hunk 7 above)
Comment (baristaGeek) [reply utils/actions/detectConsoleLogs.ts:130]: Faster yes, less prone to failing no unless we build a regex per language or use tree-sitter. It's a trade-off. 

But also think that parsing code with GPT is just the start of building our own rules engine. Is the code clean or not? Does it have too much cyclomatic complexity? Is there some redundancy? Is it sub-optimal to run that piece of code client-side? Basically, any rule that we can think of. Taking advantage of the indexed context in the future. All within the same function. 

Therefore, it's a good idea to fit it in like this (and eventually change the filename from `detecConsoleLogs.ts `to something like `parseThroughRulesEngine.ts`).

File: utils/actions/detectConsoleLogs.ts (hunk 5 above)
Comment (baristaGeek) [reply utils/actions/detectConsoleLogs.ts:168]: Ok great feedback. We can change the event to `COMMENT`, or `APPROVE`. I'll change it to the former.

Comment (EstebanDalelR) [reply utils/actions/detectConsoleLogs.ts]: Thank you!

File: utils/actions/detectConsoleLogs.ts (hunk 6 above)
Comment (EstebanDalelR) [reply utils/actions/detectConsoleLogs.ts:79]: Ok, great

File: utils/actions/detectConsoleLogs.ts (hunk 7 above)
Comment (EstebanDalelR) [reply utils/actions/detectConsoleLogs.ts:130]: Ok yeah, I see the point. 
Not a fan, but off it goes ⛵

**Review by EstebanDalelR — APPROVED** [review]

**Pull request merged**